app/src/main/java/com/example/pix/login/LoginActivity.java

```diff
@@ -88,6 +105,16 @@ public void done(ParseUser user, ParseException e) {
         // When button is hit, create an AuthenticationRequest and jump to the Spotify-provided LoginActivity
         (findViewById(R.id.auth_spotify)).setOnClickListener(unusedView -> {
 
+            // If the USer doesn't have Spotify installed, launch the download page on the Play Store
```

Comment:
Maybe you can show a message before switching to the play store? This looks a little abrupt.

Reply:
Also, can the user still use the messaging app without having Spotify installed?

Reply:
Just added this, thanks.